File: src/main/java/br/com/schumaker/musashi/crawler/model/db/repositories/MsDbFileRepository.java
package br.com.schumaker.musashi.crawler.model.db.repositories;

import br.com.schumaker.musashi.crawler.model.db.MsDbFile;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

/**
 *
 * @author Hudson Schumaker
 */
@Repository
public interface MsDbFileRepository extends MongoRepository<MsDbFile, String> {

    Page<MsDbFile> findByNameLike(String query, Pageable pageable);
    Page<MsDbFile> findByContentLike(String query, Pageable pageable);
}

File: src/main/java/br/com/schumaker/musashi/crawler/model/db/repositories/MsDbFileRepository.kt
package br.com.schumaker.musashi.crawler.model.db.repositories

import br.com.schumaker.musashi.crawler.model.db.MsDbFile
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

/**
 *
 * @author Hudson Schumaker
 */
@Repository
interface MsDbFileRepository : MongoRepository<MsDbFile?, String?> {

    fun findByNameLike(query: String?, pageable: Pageable?): Page<MsDbFile?>?
    fun findByContentLike(query: String?, pageable: Pageable?): Page<MsDbFile?>?
}